common/elasticsearch/client/v6/client_test.go

```diff
@@ -186,6 +284,74 @@ func getTestServer(t *testing.T) *httptest.Server {
 
 			w.WriteHeader(http.StatusOK)
 			w.Write([]byte(`{"count": 42}`))
+		case "/testIndex/_search":
+			w.WriteHeader(http.StatusOK)
+			w.Write([]byte(`{
+                "took": 5,
```

Comment:
instead of hardcoding different test case's expectations in the server handler you can pass these as a call back parameter to `getTestServer`. this way each test case's input and validations are self contained and test server can be more extensible for other scenarios

Reply:
Refactored, also moved the mock client initialization inside it to make the code cleaner